pkg/server/commands/listusers/list_users_rpc_test.go

```diff
@@ -595,6 +595,59 @@ func TestListUsersCycles(t *testing.T) {
 			},
 			expectedUsers: []string{},
 		},
+		{
+			name: "cycle_that_is_independent_of_tuples",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+			type user
+			type document
+				relations
+					define writer: viewer
+					define viewer: writer`,
+			tuples:        []*openfgav1.TupleKey{},
+			expectedUsers: []string{},
+		},
+		{
+			name: "cycle_when_model_has_two_parallel_edges",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "transition", Id: "1"},
+				Relation: "can_view_3",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `
+			model
+				schema 1.1
+
+			type user
+
+			type state
+				relations
+					define can_view: [user] or can_view_3 from associated_transition
+					define associated_transition: [transition]
+
+			type transition
+				relations
+					define start: [state]
+					define end: [state]
+					define can_view: can_view from start or can_view from end
+					define can_view_2: can_view
+					define can_view_3: can_view_2`,
+			tuples:        []*openfgav1.TupleKey{},
+			expectedUsers: []string{},
+		},
```

Comment:
**Request:** Can we also add a unit test for `expand` that focuses purely on the `visitedUsersets` aspect?

Reply:
sorry i don't follow, a unit test that tests which aspect?

Reply:
A unit test for the `expand` function that detects cycles solely through the visited paths attribute. Similar to what we did in https://github.com/openfga/openfga/blob/343adc7926436630e19dc8c32cc021a46af275fe/internal/graph/cycle_check_resolver_test.go#L29